examples/glue/run_glue_trainer.py

```diff
@@ -185,41 +185,62 @@ def merge_hparams(hparams: TrainerHparams, override_hparams: GLUETrainerHparams)
+        'rte': [19, 8364, 717, 10536, 90166],
+        'mrpc': [19, 8364, 717, 10536, 90166],
+        'stsb': [19, 8364, 717, 10536, 90166],
+    }
+
+    # Set up CUDA environment(s) and process pool
     ctx = mp.get_context('spawn')
     cuda_envs = init_cuda_queue(torch.cuda.device_count(), ctx)
-    finetune_tasks = list(task_to_save_ckpt.keys())
-    num_tasks = len(finetune_tasks)
-
-    if parent_ckpt:
-        wandb_group_name = parent_ckpt
-        logged_ckpt_name = parent_ckpt
-    else:
-        wandb_group_name = ckpt_load_path
-        logged_ckpt_name = ckpt_load_path
-
-    # finetuning from pretrained checkpoint
-    print(f'FINETUNING ON {ckpt_load_path}!')
-    done_callback = lambda future: log_metrics(
-        metric=future.result(), ckpt_filename=logged_ckpt_name, glue_metrics=glue_metrics)
     free_port = get_free_tcp_port()
     executor = Pool(max_workers=torch.cuda.device_count(),
                     initializer=init_cuda_env,
                     initargs=(cuda_envs, free_port),
                     mp_context=ctx)
-    for rank in range(num_tasks):
-        task = finetune_tasks[rank]
-        future = executor.submit(train_finetune, base_yaml_file, task, task_to_save_ckpt[task], ckpt_load_path,
-                                 wandb_group_name, ckpt_save_folder, save_locally, free_port + rank, load_ignore_keys)
-        future.add_done_callback(done_callback)
+
+    def make_callback(parent_ckpt, glue_metrics):
+        return lambda future: log_metrics(future.result(), ckpt_filename=parent_ckpt, glue_metrics=glue_metrics)
+
+    # Fine-tune from pre-trained checkpoint(s)
+    for parent_idx, (ckpt_load_path, parent_ckpt) in enumerate(zip(ckpt_load_paths, parent_ckpts)):
+        # `ckpt_load_path` provides the path to the checkpoint from which we load the starting weights used when fine-tuning
+        # `parent_ckpt` keeps track of the original pre-training checkpoint, for tasks with multiple fine-tuning stages (e.g., RTE)
+        # `parent_idx` is used for bookkeeping, so `parent_ckpt` can be internally recovered from the path used to save fine-tune checkpoints
+        rank = 0
+        for task, save_ckpt in task_to_save_ckpt.items():
+            if task not in seed_overrides:
+                # Run 1 fine-tune trainer from this checkpoint, with no seed overriding.
+                future = executor.submit(train_finetune, base_yaml_file, task, save_ckpt, ckpt_load_path, parent_ckpt,
+                                         parent_idx, ckpt_save_folder, save_locally, free_port + rank, load_ignore_keys)
+                future.add_done_callback(make_callback(parent_ckpt, glue_metrics))
+                rank += 1
+            else:
+                # Run multiple fine-tune trainers from this checkpoint, using a different seed override for each
+                for seed in seed_overrides[task]:
+                    future = executor.submit(train_finetune, base_yaml_file, task, save_ckpt, ckpt_load_path,
```

Comment:
should we just deepcopy `train_finetune` and update the seed here, instead of adding an extra parameter to the signature and updating the seed there?

Reply:
Given the way the `merge_hparams` function works, I'm not sure it would work to do the deepcopy strategy. My thinking was also that, because the downstream `train_finetune` function already handles several overwrites, it would be cleaner to handle the seed overwrite in that area of the code.

Reply:
That is fair, I can go ahead and resolve this one!